### PR TITLE
3307 fixes the resampling option niftisaver

### LINKED ISSUE #3307: Niftisaver doesn't save new voxelspacing
### Discussed in https://github.com/Project-MONAI/MONAI/discussions/3299 as well as https://github.com/Project-MONAI/MONAI/discussions/2029

<div type='discussions-op-text'>

<sup>Originally posted by **Gijz33** November 10, 2021</sup>
Hi,

I am changing the voxel spacing of my CT data using monai.transforms.Spacingd and saving the CT with new spacing into a .nii.gz-file using monai.data.NiftiSaver.
The voxel size transformation is succesful inside the python script. However, opening the .nii.gz file using different software (3D-slicer) the spacing of the old/orginal CT is used. Can someone help me out? A snippet of my code is shown below:

```python
train_images = sorted(glob.glob(os.path.join(data_folder, "*_ct.nii.gz")))
train_labels = sorted(glob.glob(os.path.join(data_folder, "*_seg.nii.gz")))

data_dicts = [    {"image": image_name, "label": label_name} for image_name, label_name in zip(train_images, train_labels)    ]

loader = LoadImaged(keys=("image","label"))
data_dict = loader(data_dicts[0])
     
add_channel = AddChanneld(keys=["image", "label"])
data_dict = add_channel(data_dict)

orientation = Orientationd(keys=["image", "label"], axcodes="LPS")
data_dict = orientation(data_dict)

spacing = Spacingd(keys=["image", "label"], pixdim=(0.8, 0.8, 3.0), mode=("bilinear")) 
data_dict = spacing(data_dict)


saver = NiftiSaver(output_dir="./", output_postfix="test" ,output_ext=".nii.gz",mode="nearest")
saver.save(data_dict["image"], data_dict['image_meta_dict'])
```

</div>

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #3307 

### Description
update the behaviour to skip `NiftiSaver` resampling when `resample=False`

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/nifti_saver.py [hunk 1]
```diff
@@ -131,7 +132,7 @@ def save(self, data: Union[torch.Tensor, np.ndarray], meta_data: Optional[Dict]
         """
         filename = meta_data[Key.FILENAME_OR_OBJ] if meta_data else str(self._data_index)
         self._data_index += 1
-        original_affine = meta_data.get("original_affine", None) if meta_data else None
+        original_affine = meta_data.get("original_affine", None) if meta_data and self.resample else None
```
Comment (Nic-Ma) [inline monai/data/nifti_saver.py:135]: Hi @wyli @ericspod ,

In the previous code, if `resample` is False, we only correct the orientation based on original affine and don't resample image size, does this PR change the behavior to even skip orientation? May I know why?

Thanks in advance.

Comment (wyli) [reply monai/data/nifti_saver.py:135]: At this level of saver, I feel ignoring the original affine is a more common use case.

Comment (Nic-Ma) [reply monai/data/nifti_saver.py:135]: So `resample` also includes `orientation` logic? I don't know the common practice in medical domain specific tasks, just want to double confirm with others @ericspod @rijobro .

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me.

Thanks.

**Pull request merged**